### PR TITLE
Finished conversion to Swift 4

### PR DESCRIPTION
**Pull request by Davidde94**
Conversion to Swift 4.0.3 has been finalised, with all warnings removed.

## Motivation and Context
The warnings were irritating me across several projects.

## How Has This Been Tested?
Using the already-existing test suite

## Checklist:
- [x] I have submitted a [CLA form](https://github.com/IBM-Swift/CLA)
- [x] If applicable, I have updated the documentation accordingly.
- [x] If applicable, I have added tests to cover my changes.

### CODE REVIEW

**Comment by quanvo87** [comment]
Great! I'll review and re-run Travis some time today. Travis build won't pass unless I do that. (We need to change this)

File: Sources/SwiftSMTP/Sender.swift [hunk 1]
```diff
@@ -149,7 +149,7 @@ extension String {
             throw SMTPError(.createEmailRegexFailed)
         }
 
-        let range = NSRange(location: 0, length: self.characters.count)
+        let range = NSRange(location: 0, length: self.count)
```
Comment (quanvo87) [inline Sources/SwiftSMTP/Sender.swift]: You could remove `self` in `self.count` here if you wanted.

Comment (Davidde94) [reply Sources/SwiftSMTP/Sender.swift]: Does IBM have any Swift style guidelines I could follow in the future? I understand that Swift typically advises against using "self", but didn't change it in this instance as it was used in "self.characters.count".

**Comment by quanvo87** [comment]
Looking into how to get the Travis build working again.

File: Sources/SwiftSMTP/Sender.swift (hunk 1 above)
Comment (ianpartridge) [reply Sources/SwiftSMTP/Sender.swift]: In general we've followed https://github.com/raywenderlich/swift-style-guide although it's a bit out of date now.  Removing `self.` in this case is a good idea :)

**Comment by ianpartridge** [comment]
Hmm, still failing. @quanvo87 will try another tactic...

**Comment by Davidde94** [comment]
@ianpartridge @quanvo87 you could always just take my word that it works? Who needs thorough testing, right?

**Comment by quanvo87** [comment]
That's half the fun :)

**Comment by quanvo87** [comment]
I've implemented a new dev ops flow. It is an improvement over the old one, but we still cannot get around the fact that Travis repo env vars are not available to forks (for good reason). So for now, I've pulled this PR into a PR of my own, #58. Tests pass because my PR is not a fork and Travis can access the repo env vars. I'll merge and provide credit to @Davidde94 in the release notes.

Future considerations for dev ops improvements could be:
- changing the tests to not be end, but mock the smtp protocol instead
- change any tests that are against a live smtp server to not run on forks, and when forks get merged, automatically trigger a build on the main repo (is this possible?)
- others?

**Comment by quanvo87** [comment]
https://github.com/IBM-Swift/Swift-SMTP/releases/tag/v2.0.0

**Comment by ianpartridge** [comment]
Thanks for doing this @quanvo87 !

**Comment by Davidde94** [comment]
Thanks :)

**Comment by ianpartridge** [comment]
@quanvo87 re: others? Change unit tests to use a botnet spam relay, so no auth is required?

**Pull request closed without merging**